FINERACT-2081: fix sonar issues in EntityTables enum

7ossam7atem1 opened the pull request:
## Description

Fix Sonar-reported code quality issues in EntityTables enum:
- Renamed VALUES field to ENTITY_VALUES to prevent clash with values() method
- Changed static field access modifier to package-private
- Replaced Guava ImmutableList with Java List.of()

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).


## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

adamsaghy commented on fineract-core/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/EntityTables.java:
```diff
@@ -66,15 +65,15 @@ public enum EntityTables {
         this.foreignKeyColumnNameOnDatatable = foreignKeyColumnNameOnDatatable;
         this.refColumn = refColumn;
-        this.checkStatuses = statuses == null ? ImmutableList.of() : ImmutableList.copyOf(statuses);
+        this.checkStatuses = statuses == null ? List.of() : List.copyOf(Arrays.asList(statuses));
```
Is there any reason to change from `ImmutableList` to `List`?
Also we Dont need multiple copyint... a simple `List.of(statuses)` enough

oleksii-novikov-onix replied:
I think it makes sense to use `List.of()` from Java instead of Guava's `ImmutableList`. I believe Sonar (or whatever tool) reports this as an issue. No libraries, no problems :wink:

adamsaghy commented on fineract-core/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/EntityTables.java:
```diff
@@ -135,6 +134,6 @@ public static List<Integer> getCheckStatusCodes(String name) {
     @NotNull
     public static List<EntityTables> getFiltered(Predicate<EntityTables> filter) {
-        return Arrays.stream(VALUES).filter(filter).toList();
+        return List.of(Arrays.stream(ENTITY_VALUES).filter(filter).toArray(EntityTables[]::new));
```
Same here... we dont need List.of + Arrays back and forth....

adamsaghy reviewed (changes requested):
Kindly see my review!

7ossam7atem1 commented:
> Kindly see my review!

I've reviewed and addressed all the feedback points:

- Maintained ImmutableList for checkStatuses field
- Kept the appropriate immutability approach in the constructor
- Simplified getFiltered method to avoid unnecessary List/Array conversions

Please let me know if any further changes are needed.

adamsaghy commented:
@7ossam7atem1 Kindly squash your commits!

7ossam7atem1 committed:
FINERACT-2081: fix sonar issues in EntityTables enum

7ossam7atem1 commented:
> @7ossam7atem1 Kindly squash your commits!
@adamsaghy done

adamsaghy commented:
@7ossam7atem1 There is an extra E2E test file in your PR. Would you mind to revert that one?

7ossam7atem1 commented:
> @7ossam7atem1 There is an extra E2E test file in your PR. Would you mind to revert that one?

Done
